### PR TITLE
[CodeGen] Use cached version of getRegPressureSetLimit

### PR DESCRIPTION
**Pull request by wangpc-pp**
To reduce compile time.

This is a follow-up of #118787.

### CODE REVIEW

**Comment by topperc** [comment]
Why do we need #118787 if we can just update the passes to use RegisterClassInfo?

**Review by lenary — APPROVED** [review]
LGTM for the target-independent changes.

File: llvm/lib/CodeGen/MachineLICM.cpp [hunk 1]
```diff
@@ -123,6 +123,7 @@ namespace {
     const TargetRegisterInfo *TRI = nullptr;
     const MachineFrameInfo *MFI = nullptr;
     MachineRegisterInfo *MRI = nullptr;
+    RegisterClassInfo RegClassInfo;
```
Comment (arsenm) [inline llvm/lib/CodeGen/MachineLICM.cpp:126]: The way RegisterClassInfo is currently used, it gets recomputed for every pass that uses it. It should probably move to be a normal analysis

Comment (lenary) [reply llvm/lib/CodeGen/MachineLICM.cpp:126]: I was thinking this, especially as it has some saved state to work out when it needs to recompute. I think that's probably a good follow-up?

Comment (arsenm) [reply llvm/lib/CodeGen/MachineLICM.cpp:126]: Relatedly, we also do not serialize the set of reserved registers in MIR

**Comment by wangpc-pp** [comment]
closed as it has been splitted into several small patches.

**Pull request closed without merging**